draft-irtf-panrg-path-properties.md

```diff
@@ -84,6 +84,9 @@ Note that this document does not assume that any of the listed path properties a
 
 # Terminology
 
+Entity:
+: A physical or virtual device or function which can process packets (i.e., a node or host), measure path properties, or access information about paths. With respect to a given communication (i.e., a flow), an entity may be on the data plane or control plane, and it may be on-path or off-path.
+
 Node:
 : An entity which processes packets, e.g., sends, receives, forwards, or modifies them. A node may be physical or virtual, e.g., a physical device or a service function provided as a virtual element. A node may also be the collection of multiple entities which, as a collection, processes packets, e.g., an entire Autonomous System (AS).
```

Comment:
Actually, thinking about this discrepancy a bit more:
Since we added a clear definition of entity, and a node is defined as a collection of entities, it means not every node is an entity.
In the current draft, whenever we mention "entity" in the draft, we want to include any possible node and host.
Now, however, if we talk about an "entity" we only include nodes consisting of a single entity, which might not be what we want in many places of the text.

I think we could solve this by:
- changing the entity and node definition to ensure that every node is an entity (and include in the entity definition that an entity might consist of a collection of physical or virtual devices or functions).
- whenever we talk about an "entity" in the text and want to include the definition of a node, we make sure that we say "a node or an entity". However I think this is not a very elegant solution.

Reply:
Thanks for catching this.
I think the best way forward is to change the entity and node definition to ensure that every node is an entity.
I was actually intending to make "entities" a superset of "nodes", i.e., an entity is broader than a node, because we need nodes to define a path, and an entity can be basically anything and doesn't have to be on the path.

Changing the "Entity" definition now, and took a stab at simplifying the "Node" definition as well. I don't see the need to define a Node as a collection of entities now, as an entity can already be a collection. Do you agree?

Reply:
Yes, if we phrase it in this way then it should be clear.